### ISSUE TITLE
Feature Request: Password Change : XEP-0077

### ISSUE THREAD

**Issue by philipishungry** [dino/dino#261]
Feature Request: Password Change : XEP-0077

The 'jabber:iq:register' namespace enables a user to change his or her password with a server or service. Once registered, the user can change passwords by sending an XML stanza of the following form:

Example 18. Password Change
```
<iq type='set' to='shakespeare.lit' id='change1'>
  <query xmlns='jabber:iq:register'>
    <username>bill</username>
    <password>newpass</password>
  </query>
</iq>
```
https://xmpp.org/extensions/xep-0077.html#usecases-changepw

**Comment by AlexandriaOL** [comment]
Hi, I'm bumping this and stating that this not being present has single-handedly turned me away from Dino for the foreseeable future. This stops adoption from the get-go, at least in my case, because I was handed an account and advised to change the password. Why would I use a client that does not allow me to do that?

In addition, the fact that this issue is listed under "Feature Requests" and not "Security Issues" honestly concerns me, the fact it's implementation as a feature consists of sending a single XML form, but is put in the same category as much, much more complex features and longer-term goals like "Support Audio/Video Chat", and "Windows support" makes me seriously doubt the maintainers of the project, and makes me wonder why the project does not see the lack of an ability to change your password as a security feature.

Thanks for reading.

**Comment by CryptGoat** [comment]
Another bump here. I would love to roll out Dino as the default client for my organization which exclusively runs Linux systems but not being able to change the account password is sadly a huge blocker. If I setup a new account I want the owner of the account to be able to change the password themselves. It indeed is a security concern because some people might even forget to change their password at all. Pretty much any other common XMPP client allows to change their password from their GUI. Please consider to give this issue more attention.

**Comment by mwild1** [comment]
I'm surprised (or maybe not?) that organizations are handing around passwords in this way, instead of using better methods such as LDAP (where users have the same password across services, managed elsewhere) or account creation invitations (so the user can set their own password immediately when they create the account).

I guess what I'm trying to say is: I think this is a feature worth having in Dino, but there are other (better) ways of managing user accounts in an organization, so I don't think this issue is strictly a security concern (allowing people to hand passwords around is the security concern - e.g. you can't guarantee the user will actually change it).

**Comment by CryptGoat** [comment]
> I'm surprised (or maybe not?) that organizations are handing around passwords in this way, instead of using better methods such as LDAP (where users have the same password across services, managed elsewhere) or account creation invitations (so the user can set their own password immediately when they create the account).
@mwild1 True, although it really depends on the scenario. The smaller the organization the more likely it is for it to fall back on simple, built-in options. Let's agree that the inability to change the password makes some situations much more complicated that they need to or should be.

Also, the Tails project (a [Linux distribution focused on privacy/anonymity using Tor) is still working on choosing a Pidgin replacement and right now the only candidates are Dino and Gajim: https://gitlab.tails.boum.org/tails/blueprints/-/wikis/replace_Pidgin
Right now the ability to change account passwords is not even being listed as a relevant feature but I don't think the current inability to do so would improve Dino's chances. I really would like to see the rather light-weight Dino than the kind of bloated Gajim being used in Tails.

**Comment by pukkamustard** [comment]
> instead of using better methods such as LDAP (where users have the same password across services, managed elsewhere)

I host some services (including XMPP) backed by LDAP. The primary interface for changing passwords is XMPP (via XEP-0077).

Just saying that this feature makes a lot of sense also if you are using something like LDAP.

**Comment by BeepBoopBang** [comment]
I would really appreciate this feature.

**Comment by Miss-Underst00d** [comment]
@AlexandriaOL
>not "Security Issues" honestly concerns me

Yeah, I'm done with Dino to be honest. 

Pretty sure I got some kind of exploit or buffer overflow as after joining a certain xmpp room. It glitched out and showed everyone's client in room when hovering over the room on left side as if it was a contact but it was not. After that happened next time when trying to add new contacts or even messaging existing contacts they wouldn't get the messages but I could theirs (omemo and *non omemo*). When trying to add new contact they couldn't fetch my omemo keys and would say offline (both ways) even though we added each other. 

*This was fixed by creating a new account and using profanity or gajim and uninstalling Dino.* 

>makes me seriously doubt the maintainers of the project,

Yeah they seem to care more about [social or political issues](https://dino.im/blog/2023/02/dino-0.4-release/) more then the quality of the code and client security or features.
